### PR TITLE
Elasticsearch: Removed reference to obsolete esVersion value

### PR DESCRIPTION
**Pull request by gabor**
in a previous PR ( https://github.com/grafana/grafana/pull/63341 )  we made the "elastic database version" auto-detected by grafana.

previously the user had to specify the elasticsearch version on the datasource config page.

this PR removes all the now-unnecessary fields, and values, that referred to the old functionality, also updates the documentation.

the diff is large, but most of it is updates to the tests, and deleting code from non-tests.

how to test:
1. try out the elastic datasource config page, try to create a new elastic datasource, modify it's attributes, it should work.

### CODE REVIEW

File: public/app/plugins/datasource/elasticsearch/datasource.test.ts [hunk 1]
```diff
@@ -701,7 +700,7 @@ describe('ElasticDatasource', () => {
       const { ds } = getTestContext({
         data,
         database: 'genuine.es7._mapping.response',
-        jsonData: { esVersion: '7.10.0' },
+        jsonData: {},
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/elasticsearch/datasource.test.ts:703]: @gabor Can we please remove all of these empty `jsonData` as we don;t use it to override anything and it is not required?

**Review by ivanahuckova — COMMENTED** [review]
Left 1 comment bellow.

**Comment by ivanahuckova** [comment]
Also here https://github.com/grafana/grafana/blob/gabor/es-detect-version-cleanup/pkg/tsdb/elasticsearch/elasticsearch.go#L169 we have `coerceVersion` that is only used in tests to verify its functionality. So as we don't use it, tests and also implementation can be removed.

**Comment by gabor** [comment]
> There is also left reference to esVersion in `pkg/tests/api/elasticsearch/elasticsearch_test.go` line 50.

yup, that one i'm planning to fix in a separate PR because that's owned by "not us" ( https://github.com/grafana/grafana/pull/65439 )

**Comment by gabor** [comment]
> Also here https://github.com/grafana/grafana/blob/gabor/es-detect-version-cleanup/pkg/tsdb/elasticsearch/elasticsearch.go#L169 we have `coerceVersion` that is only used in tests to verify its functionality. So as we don't use it, tests and also implementation can be removed.

good catch! removed (also the tests)

**Review by ivanahuckova — APPROVED** [review]
LGTM 🤩 
image

**Pull request merged**